src/send/saga.ts

```diff
@@ -163,44 +166,98 @@ export function* buildAndSendPayment(
-  const tokenInfo = yield* call(getTokenInfo, tokenId)
-  if (!tokenInfo) {
-    throw new Error('token info not found')
-  }
-
+export function* sendPaymentSaga({
+  amount,
+  tokenId,
+  usdAmount,
+  comment,
+  recipient,
+  fromModal,
+  preparedTransaction: serializablePreparedTransaction,
+}: SendPaymentAction) {
   try {
+    yield* call(getConnectedUnlockedAccount)
+    SentryTransactionHub.startTransaction(SentryTransaction.send_payment)
+    const context = newTransactionContext(TAG, 'Send payment')
+    const recipientAddress = recipient.address
+    if (!recipientAddress) {
+      // should never happen. TODO(ACT-1046): ensure recipient type here
+      // includes address
+      throw new Error('No address found on recipient')
+    }
+
+    const tokenInfo = yield* call(getTokenInfo, tokenId)
+    const network = getNetworkFromNetworkId(tokenInfo?.networkId)
+    if (!tokenInfo || !network) {
+      throw new Error('Unknown token network')
+    }
+    const networkId = tokenInfo.networkId
+
+    const wallet = yield* call(getViemWallet, networkConfig.viemChain[network])
+
+    if (!wallet.account) {
+      // this should never happen
+      throw new Error('no account found in the wallet')
+    }
+
     ValoraAnalytics.track(SendEvents.send_tx_start)
-    yield* call(viemSendPayment, {
-      context,
-      recipientAddress,
-      amount,
+
+    const preparedTransaction = getPreparedTransaction(serializablePreparedTransaction)
+    const feeCurrencyId = getTokenId(networkId, getFeeCurrency(preparedTransaction))
+
+    Logger.debug(
+      `${TAG}/sendPaymentSaga`,
+      'Executing send transaction',
+      context.description ?? 'No description',
+      context.id,
       tokenId,
-      comment,
-      feeInfo,
-      preparedTransaction,
-    })
+      amount
+    )
+
+    const hash = yield* call([wallet, 'sendTransaction'], preparedTransaction as any)
+
+    Logger.debug(`${TAG}/sendPaymentSaga`, 'Successfully sent transaction to the network', hash)
+
+    yield* put(
+      addStandbyTransaction({
+        __typename: 'TokenTransferV3',
+        type: TokenTransactionTypeV2.Sent,
+        context,
+        networkId,
+        amount: {
+          value: amount.negated().toString(),
+          tokenAddress: tokenInfo.address ?? undefined,
+          tokenId,
+        },
+        address: recipientAddress,
+        metadata: {
+          comment,
+        },
+        transactionHash: hash,
+        feeCurrencyId,
+      })
+    )
+
+    const receipt: TransactionReceipt = yield* call(
+      [publicClient[network], 'waitForTransactionReceipt'],
+      { hash }
+    )
+
+    Logger.debug(`${TAG}/sendPaymentSaga`, 'Got send transaction receipt', receipt)
+
+    yield* call(
+      handleTransactionReceiptReceived,
+      context.id,
+      receipt,
+      networkConfig.networkToNetworkId[network],
+      feeCurrencyId
+    )
+
+    if (receipt.status === 'reverted') {
+      throw new Error('transaction reverted')
```

Comment:
nit: would be nice to include the txHash in the error message for debugging purposes.